src/Http/Routing/src/Matching/HostMatcherPolicy.cs

```diff
@@ -48,6 +62,93 @@ public bool AppliesToEndpoints(IReadOnlyList<Endpoint> endpoints)
             });
         }
 
+        public Task ApplyAsync(HttpContext httpContext, EndpointSelectorContext context, CandidateSet candidates)
+        {
+            if (httpContext == null)
```

Comment:
@benaadams Should this error handling be moved to another method?

Reply:
This is interface dispatch. it's not going to get inlined

Reply:
Sure but there's code bloat here and nobody is going to pass null except for a unit test 😄

Reply:
Don't blame me. C# should have a feature for this.